golem-rib/src/parser/rib_expr.rs

```diff
@@ -12,38 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use combine::parser;
 use combine::parser::char;
-use combine::parser::char::{char, spaces};
-use combine::{eof, ParseError, Parser};
-use combine::{parser, sep_by};
+use combine::parser::char::spaces;
+use combine::{ParseError, Parser};
 
 use crate::expr::Expr;
 use crate::parser::errors::RibParseError;
 
 use super::binary_op::BinaryOp;
 
-// Parse a full Rib Program, and we expect the parser to fully consume the stream
-// unlike rib block expression
-pub fn rib_program<Input>() -> impl Parser<Input, Output = Expr>
```

Comment:
This can exist as a confusion and hence removed. The only entry point to parsing Rib is now Expr::from_text